core/src/solve.jl

```diff
@@ -1080,6 +1075,19 @@ function formulate_flow!(
     return nothing
 end
 
+function formulate_flow!(
+    node::AbstractParameterNode,
+    p::Parameters,
+    storage::AbstractVector,
+    t::Float64,
+)::Nothing
+    node_type = nameof(typeof(node))
+    if !isa(node, Union{Basin, LevelBoundary, DiscreteControl, PidControl, Terminal})
```

Comment:
What is the advantage of this approach over explicitly listing the `formulate_flow!` methods we want to call?

If I understand correctly, this PR switches to looping over all nodetypes, with the exception that FractionalFlow need to be last and this method that does nothing for certain allowed node types.

The only benefit I see is that you get an error when you add a node type but forgot to call `formulate_flow!`. Though that wouldn't really happen unless we add untested node types. The old method is more explicit and easier to read in my opinion, and more flexible as well in case we want to change the arguments for one of the methods.